internal/controller/runtime/fsm/runtime_fsm_select_shoot_processing.go

```diff
@@ -19,38 +19,42 @@ func sFnSelectShootProcessing(_ context.Context, m *fsm, s *systemState) (stateF
 	if s.shoot.Spec.DNS == nil || s.shoot.Spec.DNS.Domain == nil {
 		msg := fmt.Sprintf("DNS Domain is not set yet for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Info(msg)
-		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
```

Comment:
will the metrics be up to date when using `requeueAfter`? They seem to be updated only in `updateStatusAndRequeueAfter` function.

Reply:
That's correct, thank for spotting this !